src/aequitas/flow/methods/preprocessing/label_flipping.py

```diff
@@ -0,0 +1,138 @@
+from .preprocessing import PreProcessing
+
+from ...utils import create_logger
+from ...utils.imports import import_object
+
+import inspect
+import pandas as pd
+from typing import Optional, Tuple
+import numpy as np
+from sklearn.ensemble import BaggingClassifier
+
+METHODS = ["ensemble_margin", "residuals"]
+
+class LabelFlipping(PreProcessing):
+    def __init__(
+            self,
+            flip_rate: float = 0.1,
+            bagging_max_samples: float = 0.5,
+            base_estimator: str = "sklearn.tree.DecisionTreeClassifier",
+            base_estimator_args: dict = {}, 
+            n_estimators: int = 10,
+            fair_ordering: bool = True,
+            ordering_method: str = "ensemble_margin",
+            seed: int = 42
+        ):
+        """Flips the labels of a fraction of the training data according to the Fair 
+        Ordering-Based Noise Correction method.
+        
+        Parameters
+        ----------
+        flip_rate : float, optional
+            Maximum fraction of the training data to flip, by default 0.1
+        bagging_max_samples : float, optional
+            The number of samples to draw from X to train each base estimator of the 
+            bagging classifier (with replacement).
+        base_estimator : str, optional
+            The base estimator to fit on random subsets of the dataset. If None, the 
```

Comment:
If `None`, I believe this will be an error. This is a default! :stuck_out_tongue: